### PR TITLE
Update SonarLint

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**

### CODE REVIEW

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 1]
```diff
@@ -88,6 +88,7 @@
     <Rule Id="S1944" Action="Warning" />
     <Rule Id="S1994" Action="Warning" />
     <Rule Id="S2053" Action="Warning" />
+    <Rule Id="S2094" Action="Info" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:91]: S2094 - Classes should not be empty

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 2]
```diff
@@ -139,6 +140,7 @@
     <Rule Id="S2387" Action="Warning" />
     <Rule Id="S2436" Action="Warning" />
     <Rule Id="S2437" Action="Warning" />
+    <Rule Id="S2445" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:143]: S2445 - Blocks should be synchronized on read-only fields

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 3]
```diff
@@ -165,6 +167,7 @@
     <Rule Id="S2952" Action="Warning" />
     <Rule Id="S2953" Action="Warning" />
     <Rule Id="S2955" Action="Info" />
+    <Rule Id="S2970" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:170]: S2970 - Assertions should be complete

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 4]
```diff
@@ -215,6 +218,7 @@
     <Rule Id="S3366" Action="Warning" />
     <Rule Id="S3376" Action="Info" />
     <Rule Id="S3397" Action="Info" />
+    <Rule Id="S3398" Action="Info" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:221]: S3398 - "private" methods called only by inner classes should be moved to those classes

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 5]
```diff
@@ -354,6 +358,7 @@
     <Rule Id="S4462" Action="Warning" />
     <Rule Id="S4487" Action="Warning" />
     <Rule Id="S4524" Action="Warning" />
+    <Rule Id="S4545" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:361]: S4545 - "DebuggerDisplayAttribute" strings should reference existing members

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 6]
```diff
@@ -367,13 +372,16 @@
     <Rule Id="S5547" Action="Warning" />
     <Rule Id="S5659" Action="Warning" />
     <Rule Id="S5773" Action="Warning" />
+    <Rule Id="S5856" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:375]: S5856 - Regular expressions should be syntactically valid

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 7]
```diff
@@ -367,13 +372,16 @@
...
     <Rule Id="S6354" Action="None" />
     <Rule Id="S6419" Action="Warning" />
     <Rule Id="S6420" Action="Warning" />
     <Rule Id="S6421" Action="None" />
     <Rule Id="S6422" Action="Warning" />
     <Rule Id="S6423" Action="None" />
     <Rule Id="S6424" Action="Warning" />
+    <Rule Id="S6507" Action="None" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:383]: S6507 - Blocks should not be synchronized on local variables

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 8]
```diff
@@ -367,13 +372,16 @@
...
+    <Rule Id="S6513" Action="None" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:384]: S6513 - "ExcludeFromCodeCoverage" attributes should include a justification

File: analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "sonarlint.rules": {
+    "secrets:S6338": {
+      "level": "On",
+      "severity": "Blocker"
+    },
+    "secrets:S6337": {
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json:7]: S6337 - IBM API keys should not be disclosed

File: analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+      "level": "On",
+      "severity": "Blocker"
+    },
+    "secrets:S6290": {
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json:11]: S6290 - Amazon Web Services credentials should not be disclosed

File: analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      "level": "On",
+      "severity": "Blocker"
+    },
+    "secrets:S6334": {
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json:15]: S6334 - Google API keys should not be disclosed

File: analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      "level": "On",
+      "severity": "Blocker"
+    },
+    "secrets:S6336": {
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json:19]: S6336 - Alibaba Cloud AccessKeys should not be disclosed

File: analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+      "level": "On",
+      "severity": "Blocker"
+    },
+    "secrets:S6335": {
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnet_secrets_settings.json:23]: S6335 - Google Cloud service accounts keys should not be disclosed

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**